gtests/test_nopython_serializable.cpp

```diff
@@ -111,6 +111,16 @@ struct EscapeItem : SerializableItem
         register_member("escape_string", escape_string);)
 }; // end struct EscapeItem
 
+struct TestUnorderedMapItem : SerializableItem
```

Comment:
the test should be self-explained.

Reply:
Please also test for invalid key types like integer.

Reply:
The check for for invalid key types is not required, since we already have the static check, which prevents the wrong usage:
```
/home/mujin/modmesh/cpp/modmesh/serialization/SerializableItem.hpp:175:28: error: static assertion failed: Only support std::unordered_map<std::string, ...>.
  175 |         static_assert(std::is_same_v<typename T::key_type, std::string>, "Only support std::unordered_map<std::string, ...>.");
  ```

Reply:
I see.  It does not make sense to test the invalid key from C++.  But we may need to test for it from Python.  We have not had the Python wrapper for the serializer.  We should make it in a followup PR.